xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp

```diff
@@ -861,3 +861,19 @@ void CGUIDialogPVRChannelManager::Renumber()
       pItem->SetProperty("Number", "-");
   }
 }
+
+#undef BUTTON_OK
+#undef BUTTON_APPLY
+#undef BUTTON_CANCEL
+#undef RADIOBUTTON_ACTIVE
+#undef EDIT_NAME
+#undef BUTTON_CHANNEL_LOGO
+#undef IMAGE_CHANNEL_LOGO
+#undef RADIOBUTTON_USEEPG
+#undef SPIN_EPGSOURCE_SELECTION
+#undef RADIOBUTTON_PARENTAL_LOCK
+#undef CONTROL_LIST_CHANNELS
+#undef BUTTON_GROUP_MANAGER
+#undef BUTTON_NEW_CHANNEL
+#undef BUTTON_RADIO_TV
```

Comment:
Could you explain (show the exact error mesage) why this undefs are needed? Looks ugly and wrong. For example, BUTTON_RADIO_TV is only defined in this cpp, how can it cause redefinition warnings?

Reply:
> In file included from /data/src/xbmc/build/build/pvr/dialogs/CMakeFiles/pvr_dialogs.dir/Unity/unity_0_cxx.cxx:5:
/data/src/xbmc/xbmc/pvr/dialogs/GUIDialogPVRGroupManager.cpp:49:9: warning: 'BUTTON_OK' macro redefined [-Wmacro-redefined]
#define BUTTON_OK                     29
        ^
/data/src/xbmc/xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp:47:9: note: previous definition is here
#define BUTTON_OK                 4
        ^
1 warning generated.

This happens, cause unity build puts several cpp files into the same compilation unit and builds them together.
Some of those undefs might not be needed atm, but once someone defines them in another file, the issue will be back..

Reply:
Ah, I see. Another reason not to use defines. Mind if I open a PR which replaces those defines with static consts?

Reply:
that'd be great

Reply:
Given the new constants will still have "clashing" names, will the Unity build be smart enough for that?

Reply:
they must not be globals, see .e.g. android activity in this PR

Reply:
That's not what I mean. 

What happens, let's say, if I have 

static constexpr int BUTTON_OK = 4;

in GUIDialogPVRChannelManager.cpp

and 

static constexpr int BUTTON_OK = 34;

in GUIDialogPVRGroupManager.cpp ?

You said, Unity puts together several cpps in one comipilation unit... This normally would lead to a redefinition of BUTTON_OK.

Reply:
indeed, I expect it wouldn't work with unity builds.

Reply:
So, for unity builds you have to ensure that there are no symbol clashes across all source files located in the same directory? That's quite a restriction.

Reply:
It's not necessarily the whole dir, depends on UNITY_BATCH_SIZE. but yes - that's the tradeoff.

Reply:
btw, in case of defines, it's just a warning that we could just live with

Reply:
yeah, in this case defines are less evil, because changing to constexpr makes it errors.

best we can do is making the names unique across the source files I guess

Reply:
that would be the cleanest approach. I've opted for the least intrusive.
Do you want me to rename the defines instead?

Reply:
Renaming the defines would be great. I suggest the following prefixes: PVR_CHANNEL_MGR_ 
PVR_GROUP_MGR_
PVR_GUIDE_INFO_
PVR_RDS_INFO_

Reply:
> best we can do is making the names unique across the source files I guess

I guess unique across namespace works good enough, wouldn't it?
And that should be desired in anyway, imho.

Reply:
sure

Reply:
PVR defines renamed. Should I do the same for other dialogs?

another, unrelated question: any idea how to properly fix the new warning on UDFfile.h
https://jenkins.kodi.tv/job/Android-ARM/17926/clang/#issuesContent

Reply:
The forward declaration should ptobably get changed to `struct udfread`